eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -38,26 +42,33 @@ steps:
 
   - template: /eng/common/TestResources/setup-environments.yml
 
-  - pwsh: |
-      eng/common/scripts/Import-AzModules.ps1
-
-      $subscriptionConfiguration = @'
-        ${{ parameters.SubscriptionConfiguration }}
-      '@ | ConvertFrom-Json -AsHashtable;
-
-      # The subscriptionConfiguration may have ArmTemplateParameters defined, so
-      # pass those in via the ArmTemplateParameters flag, and handle any
-      # additional parameters from the pipelines via AdditionalParameters
-      eng/common/TestResources/New-TestResources.ps1 `
-        -ResourceType '${{ parameters.ResourceType }}' `
-        -ServiceDirectory '${{ parameters.ServiceDirectory }}' `
-        -Location '${{ parameters.Location }}' `
-        -DeleteAfterHours '${{ parameters.DeleteAfterHours }}' `
-        @subscriptionConfiguration `
-        -AdditionalParameters ${{ parameters.ArmTemplateParameters }} `
-        -CI `
-        -Force `
-        -Verbose | Out-Null
+  - task: AzurePowerShell@5
     displayName: Deploy test resources
     env:
       TEMP: $(Agent.TempDirectory)
+    inputs:
+      azureSubscription: ${{ parameters.ServiceConnection }}
+      azurePowerShellVersion: LatestVersion
+      pwsh: true
+      ScriptType: InlineScript
+      Inline: |
+        eng/common/scripts/Import-AzModules.ps1
+
+        $subscriptionConfiguration = @'
+          ${{ parameters.SubscriptionConfiguration }}
```

Comment:
Do we need to delete or remove some of the existing SP credentials from the configurations?

Reply:
In the case of `-UseFederatedAuth` any secrets passed to the script are NOT passed to the templates or pre-/post- scripts. 

This way we don't have to make changes to sub-configs until all services have migrated.

Reply:
Yes the plan here was not to change the sub configs or introduce layers as its a larger change. Trying to keep each incremental step small here so we can start unblocking sdk teams who have DAC changes on the way.